pkg/replicaset/replicaset_test.go

```diff
@@ -0,0 +1,607 @@
+			apiClientNil:        true,
+		},
+		{
+			name:                defaultReplicaSetName,
+			namespace:           "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("replicaset 'nsname' cannot be empty"),
+			apiClientNil:        true,
+		},
+		{
+			name:                "replicaset-test",
+			namespace:           defaultReplicaSetNamespace,
+			addToRuntimeObjects: false,
+			expectedError: fmt.Errorf("replicaset object replicaset-test does not exist " +
+				"in namespace test-namespace"),
+			apiClientNil: true,
+		},
+		{
+			name:                defaultReplicaSetName,
+			namespace:           defaultReplicaSetNamespace,
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("replicaset 'apiClient' cannot be empty"),
+			apiClientNil:        false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testReplicaSet := generateReplicaSet(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testReplicaSet)
+		}
+
+		if testCase.apiClientNil {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := Pull(testSettings, testCase.name, testCase.namespace)
+
+		if testCase.expectedError != nil {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		} else {
+			assert.Equal(t, testReplicaSet.Name, builderResult.Object.Name)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestNewReplicaSetBuilder(t *testing.T) {
+	testCases := []struct {
+		name          string
+		namespace     string
+		labels        map[string]string
+		container     []corev1.Container
+		expectedError string
+	}{
+		{
+			name:          defaultReplicaSetName,
+			namespace:     defaultReplicaSetNamespace,
+			labels:        defaultReplicaSetLabel,
+			container:     defaultReplicaSetContainer,
+			expectedError: "",
+		},
+		{
+			name:          "",
+			namespace:     defaultReplicaSetNamespace,
+			labels:        defaultReplicaSetLabel,
+			container:     defaultReplicaSetContainer,
+			expectedError: "replicaset 'name' cannot be empty",
+		},
+		{
+			name:          defaultReplicaSetName,
+			namespace:     "",
+			labels:        defaultReplicaSetLabel,
+			container:     defaultReplicaSetContainer,
+			expectedError: "replicaset 'nsname' cannot be empty",
+		},
+		{
+			name:          defaultReplicaSetName,
+			namespace:     defaultReplicaSetNamespace,
+			labels:        map[string]string{},
+			container:     defaultReplicaSetContainer,
+			expectedError: "replicaset 'labels' cannot be empty",
+		},
+		{
+			name:          defaultReplicaSetName,
+			namespace:     defaultReplicaSetNamespace,
+			labels:        defaultReplicaSetLabel,
+			container:     []corev1.Container{},
+			expectedError: "replicaset 'containerSpec' cannot be empty",
+		},
```

Comment:
Please add client boolean param and add test case with:

			name:          defaultReplicaSetName,
			namespace:     defaultReplicaSetNamespace,
			labels:        defaultReplicaSetLabel,
			container:     defaultReplicaSetContainer,
			expectedError: "",
                        client: false